### PR TITLE
Develop sign up functionality

### LINKED ISSUE #3: Develop sign up screen functionality
## REQUIREMENTS / DESCRIPTION

```[tasklist]
### Tasks
- [ ] Use useState hook to add functionality to fields
- [ ] Use the database connection to let the application save new users
- [ ] Create the session
- [ ] Add validation to email field
- [ ] Add validation to name field
- [ ] Add validation to password field
```

### PR DESCRIPTION
**Pull request by BrandonVG**
# Description

### Project change
This branch has a lot of changes but they are actually not real changes, we had a problem with the first generated project so we generated a new project with CLI and then just added the old changes we did before.

So please, ignore the changes in those files and focus on the functionalities changes and files that we have in /src
### Functionalities changes
We implemented new functional changes, using the Firebase connection we did in the past to make functional the sign-up screen, working to create users with email and password or using a Google account, we also added some validations to the form to only accept valid emails and passwords, also avoiding empty texts, for now, this signup just create the user in FireBase because we don't have another screen to redirect after this.

Finally, other functionalities like the password reveal on the icon eye pressed or the buttons being disabled until the user accepts the checkbox was added 

### Project change
We needed to create a new base project because the first one gave us some problems with google sign in because of some SHA1 problems so by recreating the project we solved these problems, maybe this solution was not the best solution, but it worked well
### Function changes
We needed these changes to provide the user with the sign-in function and have the functionalities required in the requirements

## Screenshots

image

Co-authored-by: Tole15 <[rodrigermoon@gmail.com](mailto:rodrigermoon@gmail.com)>
Co-authored-by: Axw3ll <[axelomar.a.v@gmail.com](mailto:axelomar.a.v@gmail.com)>

### CODE REVIEW

**Review by Eduardoespl — APPROVED** [review]
💯

File: .watchmanconfig [hunk 1]
```diff
@@ -1 +1 @@
-{}
\ No newline at end of file
+{}
```
Comment (ecruz17) [inline .watchmanconfig:1]: nice

File: src/hooks/SignInWithGoogle.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import auth from '@react-native-firebase/auth';
+import {GoogleSignin} from '@react-native-google-signin/google-signin';
+
+GoogleSignin.configure({
+  scopes: ['email', 'profile'],
+  webClientId:
+    '569311043643-h9c0ctlutkqsnr2ticidfl3d08nuafqj.apps.googleusercontent.com',
```
Comment (ecruz17) [inline src/hooks/SignInWithGoogle.js]: you could place this inside an env variable

**Review by ecruz17 — APPROVED** [review]
Nice work! 🔥

**Review by AlbertoG22 — APPROVED** [review]

**Pull request merged**